simphony/engine/tests/test_engine_metadata.py

```diff
@@ -0,0 +1,134 @@
+"""Tests regarding loading engine's metadata."""
+import sys
+import unittest
+
+import simphony.engine as engine_api
+from simphony.engine import ABCEngineExtension, EngineInterface
+from simphony.engine.extension import EngineManager, EngineManagerException
+from simphony.engine.extension import EngineFeatureMetadata, EngineMetadata
+
+
+class _Example1(ABCEngineExtension):
+    def get_supported_engines(self):
+        example_engine = \
+            self.create_engine_metadata('EXAMPLE1',
+                                        None,
+                                        [EngineInterface.Internal,
+                                         EngineInterface.FileIO])
+
+        return [example_engine]
+
+    def create_wrapper(self, cuds, engine_name, engine_interface):
+        if engine_name == 'EXAMPLE1':
+            pass
+        else:
+            raise Exception('Only EXAMPLE1 engine is supported. '
+                            'Unsupported eninge: %s', engine_name)
+
+
+def get_example_engine_extension():
+    class _Example2(ABCEngineExtension):
+        def get_supported_engines(self):
+            example_engine = \
+                self.create_engine_metadata('EXAMPLE2',
+                                            None,
+                                            [EngineInterface.Internal,
+                                             EngineInterface.FileIO])
+
+            return [example_engine]
+
+        def create_wrapper(self, cuds, engine_name, engine_interface):
+            if engine_name == 'EXAMPLE2':
+                pass
+            else:
+                raise Exception('Only EXAMPLE2 engine is supported. '
+                                'Unsupported eninge: %s', engine_name)
+    return _Example2
+
+
+class TestEnginePublicAPI(unittest.TestCase):
+    """Test everything engine metadata."""
+    def setUp(self):
+        self.manager = engine_api._ENGINE_MANAGER
+
+    def tearDown(self):
+        pass
+
+    def test_get_supported_engines(self):
+        supported = engine_api.get_supported_engines()
+        assert(isinstance(supported, list))
+
+    def test_create_wrapper(self):
+        pass
```

Comment:
If this is to be completed, can you please add a FIXME/TODO comment here?  It would be easier to track these TODO's down later.